bors.toml

```diff
@@ -7,5 +7,6 @@ status = [
 	'bors-ok',
 ]
 pr_status = [
-	'linear-history'
+	'linear-history',
+	'lint',
```

Comment:
For my own edification, how does `pr_status` work? The [bors docs](https://bors.tech/documentation/#configuration-borstoml) don't really explain it well.

I see we have a couple PR checks with "linear-history" in the name:

<img width="850" alt="Screen Shot 2023-01-19 at 4 23 28 PM" src="https://user-images.githubusercontent.com/710598/213591154-c3d25ccf-dc0a-4a43-a2a4-e6ffa919a5f9.png">

<img width="831" alt="Screen Shot 2023-01-19 at 4 23 42 PM" src="https://user-images.githubusercontent.com/710598/213591185-646accc7-0249-43a1-9a6a-c079a98b2c7f.png">

Is bors looking at these? Which one?

Similarly, we have a bunch of checks with "lint" in their name. Which one is bors looking for if we specify `'lint'` in `pr_status`?

<img width="840" alt="Screen Shot 2023-01-19 at 4 25 05 PM" src="https://user-images.githubusercontent.com/710598/213591347-e3f8383e-6b2f-41d8-94df-2ede4738f503.png">

Reply:
I had cobbled this together quickly yesterday based on my limited information at the time and was initially going to see if it worked via trial and error, but it was the end of the day and I decided to put of until today to put more thought into it.

I was planning on using the documented SQL-like wildcard matching, but https://github.com/bors-ng/bors-ng/issues/1346 indicates that it's not available on `pr_status`.

I've been looking into the file history and it appears that copying the whole name of a job has worked in the past.
- https://github.com/pulumi/pulumi/blame/6f21c0dacafdcd5f7a45147ed9e192281932e9c3/bors.toml

If I require lints to pass, it may prevent merging community PRs since lint checks don't run on them unless we `/run-acceptance-tests` and I'm unsure if bors will notice those checks.

I've added the following checks to the pr_status:
```
CI / lint / Lint GHA	pass	1m2s	https://github.com/pulumi/pulumi/actions/runs/3970818388/jobs/6807021525
CI / lint / Lint Go	pass	4m38s	https://github.com/pulumi/pulumi/actions/runs/3970818388/jobs/6807021385
CI / lint / Lint Protobufs	pass	37s	https://github.com/pulumi/pulumi/actions/runs/3970818388/jobs/6807021634
CI / lint / Lint SDKs	pass	1m50s	https://github.com/pulumi/pulumi/actions/runs/3970818388/jobs/6807021722
```

I got the output of these names from using `gh pr checks 11924`